nix/tests/vm-tests/all.nix

```diff
@@ -0,0 +1,256 @@
+    # --------------------------------------------------------------------------------------------------------------
+
+    pab = { pkgs, ... }: {
+      imports = [ ../../modules/pab.nix ];
+      environment.systemPackages = with pkgs; [ curl ];
+
+      networking = {
+        firewall.allowedTCPPorts = [ 8080 8081 8082 8083 8084 8085 ];
+        dhcpcd.enable = false;
+        interfaces.eth1.ipv6.addresses = lib.mkOverride 0 [{ address = "fd00::1"; prefixLength = 64; }];
+        interfaces.eth1.ipv4.addresses = lib.mkOverride 0 [{ address = "192.168.1.1"; prefixLength = 24; }];
+      };
+
+      services.pab = {
+        enable = true;
+        pab-package = plutus-pab.pab-exes.plutus-pab;
+        contracts = [ "${marlowe-app}/bin/marlowe-app" ];
+        staticContent = marlowe-dashboard.client;
+        dbFile = "/var/lib/pab/pab-core.db";
+        defaultWallet = 1;
+        webserverPort = 8080;
+        walletPort = 8081;
+        nodePort = 8082;
+        chainIndexPort = 8083;
+        signingProcessPort = 8084;
+        metadataPort = 8085;
+      };
+    };
+
+    # ---------------------------------------------------------------------------------------------------------------
+    # playgrounds : 192.168.1.2 - running plutus/marlowe playgrounds and nginx
+    # --------------------------------------------------------------------------------------------------------------
+
+    playgrounds = { pkgs, ... }: {
+      imports = [
+        ../../modules/plutus-playground.nix
+        ../../modules/marlowe-playground.nix
+      ];
+
+      networking = {
+        firewall.allowedTCPPorts = [ 8080 9090 ];
+        extraHosts = ''
+          127.0.0.1 plutus-playground
+          127.0.0.1 marlowe-playground
+          127.0.0.1 marlowe-dashboard
+          192.168.1.1 pab
+          192.168.1.3 webghc
+        '';
+        dhcpcd.enable = false;
+        interfaces.eth1.ipv6.addresses = lib.mkOverride 0 [{ address = "fd00::2"; prefixLength = 64; }];
+        interfaces.eth1.ipv4.addresses = lib.mkOverride 0 [{ address = "192.168.1.2"; prefixLength = 24; }];
+      };
+
+      services = {
+        marlowe-playground = {
+          enable = true;
+          port = 4001;
+          playground-server-package = marlowe-playground.server;
+        };
+
+        plutus-playground = {
+          enable = true;
+          port = 4000;
+          playground-server-package = plutus-playground.server;
+          webghcURL = "http://webghc";
+        };
+
+        nginx = {
+          enable = true;
+          recommendedGzipSettings = true;
+          recommendedProxySettings = true;
+          recommendedOptimisation = true;
+
+          upstreams = {
+            plutus-playground.servers."127.0.0.1:4000" = { };
+            marlowe-playground.servers."127.0.0.1:4001" = { };
+            marlowe-dashboard.servers."192.168.1.1:8080" = { };
+          };
+          virtualHosts = {
+            "marlowe-dashboard" = {
```

Comment:
We can't include these snippets from the normal setups for the various apps, can we?

Reply:
eventually yes - not now though. These NixOS services need to first find their way into the actual configs for production.